### PR TITLE
Add compile_args to deep ensemble model

### PR DESCRIPTION
**Pull request by nfergu**
## Summary

This PR adds a `compile_args` option to the `DeepEnsemble` class. This enables various options, like JIT compilation for example, to be passed through to Keras's `compile` method.

**Fully backwards compatible:** yes

## PR checklist

- [] The quality checks are all passing
- [X] The bug case / new feature is covered by tests
- [X] Any new features are well-documented (in docstrings or notebooks)

All the tests in `tests/unit/models/keras` `tests/unit/models/gpflux` seem to pass locally and I have run `tox -e reformat` and `tox -e types` locally.

### CODE REVIEW

File: trieste/models/keras/models.py
Comment (uri-granta) [inline trieste/models/keras/models.py]: (following is shorter, but possibly less clear?)

```suggestion
        if not {"optimizer", "loss", "metrics"}.isdisjoint(compile_args):
```

File: tests/unit/models/gpflux/test_models.py
Comment (uri-granta) [inline tests/unit/models/gpflux/test_models.py]: Test against non-trivial compile args maybe?

```suggestion
        DeepGaussianProcess(mock_model, compile_args={"run_eagerly": True, "optimizer": unittest.mock.MagicMock()})
```

**Review by uri-granta — APPROVED** [review]
LGTM! Tell me once you're happy and I'll merge it.

**Comment by nfergu** [comment]
> LGTM! Tell me once you're happy and I'll merge it.

@uri-granta Thanks. I committed your suggestions and applied them a bit more widely in [this commit](https://github.com/secondmind-labs/trieste/pull/816/commits/6d25215b895fca24b48cc70d0719b4bb13513620).

I'm happy with this, assuming all the checks pass now.

**Comment by nfergu** [comment]
@uri-granta I think you need to kick off the checks on this?

**Pull request merged**
